### ISSUE TITLE
HAL_FLASHEx_OB_DBankConfig _always_ returns HAL_ERROR

### ISSUE THREAD

**Issue by daantimmer** [STMicroelectronics/stm32g4xx_hal_driver#11]
https://github.com/STMicroelectronics/stm32g4xx_hal_driver/blob/fa6618be0a5b3b918bda9fbf990c70da966f7543/Src/stm32g4xx_hal_flash_ramfunc.c#L130C30-L130C56

`status` is set to `HAL_ERROR` on line 133. But it is never set to HAL_OK when succeeded, thus always returning `HAL_ERROR`

**Comment by TOUNSTM** [comment]
Hello @daantimmer,

Thank you for your contribution. This is normal behavior. In the case of a successful `DBANK `modification, there is no output from the function because a reboot occurs to account for the new value of the option byte.

If the code execution continues past the point where a reset was expected, it implies that the programming operation was not successful, Therefore, the function returns `HAL_ERROR`. This return value indicates that an error occurred during the operation, which prevented the option byte from being successfully modified and the expected system reset from taking place.

In summary, the function is designed to return a status only if there is an error. If the operation is successful, the microcontroller resets, and the function does not return.

I hope this answers your point. Please allow me to close this issue.

With regards,
